### PR TITLE
Use .transform() instead of .setTransform() when rendering image or text

### LINKED ISSUE #14478: Rotated vector layer text styles not displaying at correct location when context is clipped and translated
**Describe the bug**
We're working on a modified spyglass using the latest version of OL, but instead of showing hidden layers centered around the cursor, we display them in a translated area above and to the right of the cursor. 

The issue is that when the context for vector layers containing _rotated_ text styles are clipped and translated, the location of these styles appear at the _original_ location on the map, not the _translated_ location. This does not occur for non-rotated text styles.

**To Reproduce**
1. Go to https://codesandbox.io/s/ol7-rotate-translate-issue-qzext2?file=/src/index.js
2. Hover the mouse over Miami. The blue box shows the spyglass area. The red box shows the hidden layers revealed by the spyglass, which are clipped and translated via prerenders and postrenders. Note that both the blue "1" and _non-rotated_ red "2" vector layer text styles are properly displayed over Miami, Florida.

image

3. A third vector layer text style - a green "3" - has a non-zero rotation value. It's also being clipped and translated should display on top of the red "2". However, it incorrectly displays in the original non-translated location of Miami from the main map. Any rotation value != 0 causes the translation to fail. 

image

**Expected behavior**
The green "3" should be translated and superimposed on top of the red "2". I've only been able to accomplish this by removing the rotation option (~Line 106 of index.js) and/or setting the value to 0.

image

### PR DESCRIPTION
**Pull request by mike-000**
Fixes #14478

Allows rotated images and text, and line placement text to be displayed offset as expected following a context translate() or scale() in a prerender event.  No rendering tests broken by this change.

Also fix similar issue in immediate renderer.

### CODE REVIEW

**Comment by kevmcgrath** [comment]
Hi @mike-000,

We're eager for this fix, which also addresses #14478 and see that it didn't make it into v7.3. Do you know if there are plans to include it in the next release?  The issue is unfortunately holding up one of our projects.

Kind Regards,
Kevin

**Review by ahocevar — CHANGES_REQUESTED** [review]
@mike-000 Thanks for this fix. I think what changes for users now is that `postrender` needs to reset any transforms that `prerender` sets. It would be good to test if this is indeed the case, and if so, add at least a paragraph in the release notes.

**Comment by mike-000** [comment]
> @mike-000 Thanks for this fix. I think what changes for users now is that `postrender` needs to reset any transforms that `prerender` sets. It would be good to test if this is indeed the case, and if so, add at least a paragraph in the release notes.

In most cases that would already been required - in `drawImageOrLabel()` `setTransform` was called inside a save/restore just as `transform` is now.  Only if the immediate renderer had called `context.setTransform(1, 0, 0, 1, 0, 0)` to revert a previous `setTransform` - which only happened if the style had scale or rotation - would it not have been necessary, but in that case the transform set by the application would not have had the expected effect on the rendering.  For example set the CircleStyle scale to anything other than 1 in https://codesandbox.io/s/feature-move-animation-forked-6vt3o9 and the translated copy of the feature does not appear.

**Comment by ahocevar** [comment]
@mike-000 Thanks for this fix. Now that we've some backwards incompatible changes lined up already, could you maybe rebase this pull request and move the upgrade notes into the "backwards incompatible changes" section? Thanks in advance!

**Review by ahocevar — APPROVED** [review]
Thanks, @mike-000

**Comment by kevmcgrath** [comment]
Thank you, @mike-000 and @ahocevar!

**Comment by tschaub** [comment]
@kevmcgrath - you may know this already, but we create a `dev` tagged release after every merge to the default branch.  So you can get this fix with `npm install ol@dev`.  As mentioned above, we have rolled in some other backwards incompatible changes, so the next stable release will be 8.0.  See the [upgrade notes](https://github.com/openlayers/openlayers/blob/main/changelog/upgrade-notes.md) for changes you might need to know about that aren't yet in a stable release.

**Pull request merged**
